### ISSUE TITLE
Use of vapor pressure deficit as a parameter in fire spread?

### ISSUE THREAD

**Issue by cjm23** [LANDIS-II-Foundation/Extension-SCRPPLE#146]
This would be a potential enhancement, but would it be possible to include VPD as a parameter for fire spread or max fire spread?  Some studies have found it useful in explaining wildfire trends in CA (e.g., Williams et al. 2019, 10.1029/2019EF001210).

**Comment by rmscheller** [comment]
Charles,
It's a good idea.  It would need to be optional as not all climate forecasts include daily VPD.  Do you know how often it is provided?

**Comment by ZacharyRobbins** [comment]
Including it as a predictor in the glm of the spread model would allow for it to be optional, it could just be set as 0.00 if you do not include it. I will research this implementation.

**Comment by cjm23** [comment]
Apologies for not following up sooner about this, the code that I have is passed on from Abatzaglou's calculation of VPD used in gridMET.  Because it uses min and max temps and RHs, it's possible to calculate a number of different VPD values (e.g., min-min, max-max, avg-avg, etc.) and I'm not sure what is "best".
      calcES <- function(temperatureFilename) {
        if (!file.exists(temperatureFilename)) {
          return(NULL)
        }
        temperatureData <- raster(temperatureFilename)
        esData <- 6.112 * exp ((17.67 * temperatureData) / (temperatureData + 243.5));
        return(esData)
      }
      
      calcVPD <- function(rhData, esData, outputFilename) {
        if (is.null(rhData) || is.null(esData)) {
          return(NULL)
        }
        vpdData <- (esData/10) * (1-rhData/100)
        writeRaster(vpdData, filename=outputFilename,
                    format='GTiff', options=c("TILED=YES","COPY_SRC_OVERVIEWS=YES","COMPRESS=LZW"),
                    overwrite=T)
        return(vpdData)
      }

**Comment by ZacharyRobbins** [comment]
Aim for June progress

**Comment by ZacharyRobbins** [comment]
@rmscheller  is VPD in the climate library? I don't mind adding it to spread, but it seems like the calculation should be done there not in the fire submodule.

**Comment by rmscheller** [comment]
Yes, to the best of my knowledge, it is an option in the climate library.
